finetune/util/huggingface_interface.py

```diff
@@ -127,28 +130,53 @@ def __init__(self):
 
         @property
         def vocab_size(self):
-            self.tokenizer.vocab_size
+            return self.tokenizer.vocab_size
 
         def _encode(self, texts):
             batch_tokens = []
             batch_token_idxs = []
             batch_char_ends = []
             batch_char_starts = []
             for i, text in enumerate(texts):
-                encoded = self.tokenizer._tokenizer.encode(
-                    text, add_special_tokens=False
-                )
-                batch_tokens.append(encoded.tokens)
-                batch_token_idxs.append(encoded.ids)
-                token_ends = []
-                token_starts = []
-                for start, end in encoded.offsets:
-                    token_starts.append(start)
-                    token_ends.append(end)
-
-                batch_char_ends.append(token_ends)
-                batch_char_starts.append(token_starts)
-
+                if self.tokenizer.is_fast:
+                    encoded = self.tokenizer._tokenizer.encode(
+                        text, add_special_tokens=False
+                    )
+                    batch_tokens.append(encoded.tokens)
+                    batch_token_idxs.append(encoded.ids)
+                    token_ends = []
+                    token_starts = []
+                    for start, end in encoded.offsets:
+                        token_starts.append(start)
+                        token_ends.append(end)
+
+                    batch_char_ends.append(token_ends)
+                    batch_char_starts.append(token_starts)
+                else:
```

Comment:
Awesome, thank you for taking the time to support the non-fast tokenizers w/o char offsets here

Reply:
Of course! In retrospect I should have done this first haha....